### PR TITLE
[UNR-2887] Fix: Dynamic subobjects attached near actor creation would not be able to replicate their properties

### PR DESCRIPTION
**Pull request by mironec**
#### Description
Previously, if trying to create and attach a dynamic subobject to an actor near its creation point, the fields of the dynamic component would not replicate properly after initial creation.
The reason for this was that we were not setting component write ACLs properly for these subobjects. This was due to the change made earlier, to resolve dynamic components earlier, which caused the code trying to check if the subobject is dynamically attached to think it was not dynamically attached, since it has already been resolved. This is fixed by decoupling the writing of ACLs from the resolution of the dynamic object in `SpatialSender::CreateEntity`.

Also a drive-by fix for $lld spam in logs.
Also a drive-by fix for authority queuing not checking which component we got authority over and potentially firing component updates for objects we did not have authority over.

#### Release note
This actually worked prior to the previous fix to dynamic subobjects (also in this RC), so I think no release note is necessary.

#### Tests
Augmented dynamic component tests: https://github.com/improbable/UnrealGDKEngineNetTest/pull/8.

#### Documentation
N/A

#### Primary reviewers
@improbable-valentyn @Vatyx

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -538,7 +538,7 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 	if (AActor* EntityActor = Cast<AActor>(PackageMap->GetObjectFromEntityId(EntityId)))
 	{
 		UE_LOG(LogSpatialReceiver, Log, TEXT("Entity for actor %s has been checked out on the worker which spawned it or is a singleton linked on this worker. "
-			"Entity id: $lld"), *EntityActor->GetName(), EntityId);
+			"Entity id: %lld"), *EntityActor->GetName(), EntityId);
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: drive-by fix

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 2]
```diff
@@ -539,15 +539,22 @@ void USpatialSender::SendComponentUpdates(UObject* Object, const FClassInfo& Inf
...
 {
 	if (TArray<Worker_ComponentUpdate>* UpdatesQueuedUntilAuthority = UpdatesQueuedUntilAuthorityMap.Find(EntityId))
 	{
-		for (Worker_ComponentUpdate& Update : *UpdatesQueuedUntilAuthority)
+		for (auto It = UpdatesQueuedUntilAuthority->CreateIterator(); It; It++)
+		{
+			if (ComponentId == It->component_id)
+			{
+				Connection->SendComponentUpdate(EntityId, &(*It));
+				It.RemoveCurrent();
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: No memory corruption?

**Comment by mironec** [comment]
By the way, I am pretty sure my previous early-resolving of dynamic subobjects breaks dynamic subobjects for which you turn replication on and off :(

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: `RemoveCurrentSwap`

Comment (mironec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Have had to replace the entire code section, now it's kind of less readable. Suggestions welcome

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 3]
```diff
@@ -292,21 +292,21 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
...
 			const FClassInfo& SubobjectInfo = ClassInfoManager->GetOrCreateClassInfoByObject(Subobject);
 
 			FRepChangeState SubobjectRepChanges = Channel->CreateInitialRepChangeState(Subobject);
 			FHandoverChangeState SubobjectHandoverChanges = Channel->CreateInitialHandoverChangeState(SubobjectInfo);
 
+			ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
+			{
+				if (SubobjectInfo.SchemaComponents[Type] != SpatialConstants::INVALID_COMPONENT_ID)
+				{
+					ComponentWriteAcl.FindOrAdd(SubobjectInfo.SchemaComponents[Type]) = AuthoritativeWorkerRequirementSet;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]:
```suggestion
					ComponentWriteAcl.Add(SubobjectInfo.SchemaComponents[Type], AuthoritativeWorkerRequirementSet);
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 2 above)
Comment (mironec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp]: Replaced back to using an iterator, because we want to preserve order.
